### PR TITLE
[VS Incentives] priviledged CreateGroupAsIncentivesModuleAcc for upgrade handler

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #6603

## What is the purpose of the change

This introduces a new exported method `CreateGroupAsIncentivesModuleAcc` to be called in the upgrade handler for creating groups.

It differs from the standard `CreateGroup` in the following way:
- Does not allow providing coins or creator address - always creates Group as incentives module account with empty coins
- Does not sync weights - this is to allow Group creation at the time of feature launch. At that time, no pool has volume they would all fail to create a Groups if called via standard `CreateGroup`
- Creation fee is not charged (since incentives module account)

To enable this, I extracted a shared internal method `createGroup` that is reused by `CreateGroup` and `CreateGroupAsIncentivesModuleAcc`

Since the diff was getting out of control, I made the decision to move all `Group` specific methods into a separate file. Will mark areas that stay the same via comments

## Testing and Verifying

- Introduced new group creation tests for each new method
- Shared as much common test logic is possible
- Attempted to highlight the differences by custom validation in tests
- Cleaned up and added test helpers

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/incentives/keeper/gauge.go [hunk 1]
```diff
@@ -241,103 +239,6 @@ func (k Keeper) CreateGauge(ctx sdk.Context, isPerpetual bool, owner sdk.AccAddr
...
-// in the pool incentives as defined by the DistrRecord. The Group's Gauge can either be perpetual or non-perpetual.
-// If numEpochPaidOver is 0, then the Group's Gauge is perpetual. Otherwise, it is non-perpetual.
-// Returns nil on success.
-// Returns error if:
-// - given pool IDs slice is empty or has 1 pool only
-// - fails to initialize gauge information for every pool ID
-// - fails to send coins from owner to the incentives module for the Group's Gauge
-// - fails to charge group creation fee
-// - fails to set the Group's Gauge to state
-func (k Keeper) CreateGroup(ctx sdk.Context, coins sdk.Coins, numEpochPaidOver uint64, owner sdk.AccAddress, poolIDs []uint64) (uint64, error) {
```
Comment (p0mvn) [inline x/incentives/keeper/gauge.go:258]: Note: all Group related methods are moved to `group.go` (same as tests)

File: x/incentives/keeper/group.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+		}
+	}
+
+	// Charge fee
+	groupCreationFee := params.GroupCreationFee
+	if err := k.bk.SendCoinsFromAccountToModule(ctx, sender, distrtypes.ModuleName, groupCreationFee); err != nil {
+		return false, err
+	}
+	return true, nil
+}
```
Comment (p0mvn) [inline x/incentives/keeper/group.go:196]: Note: These did not change. Moved over from another file

File: x/incentives/keeper/group_test.go [hunk 3]
```diff
@@ -0,0 +1,671 @@
...
+			} else {
+				s.FailNow("unknown system under test type")
+			}
+		})
+	}
+	// Ensure that we ran all the tests to prevent false positions
+	s.Require().Equal(int(TotalTypes), differentExecutionTypeCount)
+}
+
+func (s *KeeperTestSuite) runCreateGroupTests(poolInfo apptesting.SupportedPoolAndGaugeInfo, concentratedGaugeRecord types.InternalGaugeRecord, balancerGaugeRecord types.InternalGaugeRecord, stableSwapGaugeRecord types.InternalGaugeRecord, expectedCommunityPoolFeeBalance sdk.Coins, expectedGroupGaugeId uint64, originalCommunityPoolBalance sdk.Coins) {
```
Comment (p0mvn) [inline x/incentives/keeper/group_test.go:242]: Note: most of the logic in this method is the same as the old `TestCreateGroup`. The original was refactored for reuse with other group creator method tests

File: x/incentives/keeper/group_test.go [hunk 4]
```diff
@@ -0,0 +1,671 @@
...
+			if didChargeFee {
+				s.Require().Equal(senderBalanceBefore.Sub(customGroupCreationFee).String(), senderBalanceAfter.String())
+				s.Require().Equal(communityPoolBalanceBefore.Add(customGroupCreationFee...).String(), communityPoolBalanceAfter.String())
+			} else {
+				s.Require().Equal(senderBalanceBefore.String(), senderBalanceAfter.String())
+				s.Require().Equal(communityPoolBalanceBefore.String(), communityPoolBalanceAfter.String())
+			}
+		})
+	}
+}
```
Comment (p0mvn) [inline x/incentives/keeper/group_test.go:613]: Note: these are copied from another test file and are not changed

File: x/incentives/keeper/group_test.go [hunk 5]
```diff
@@ -0,0 +1,671 @@
...
+	}
+)
+
+// We have 3 different implementations of CreateGroup:
+// - CreateGroup - standard client-facing
+// - CreateGroupAsIncentivesModuleAcc - only callable by incentives module account
+// - CreateGroupInternal - internal, used by the other two
+// This test validates each of these implementations.
+// While they are similar in functionality, they still differ.
+// As a result, we crate this shared test case to avoid code duplication but
```
Comment (czarcas7ic) [inline x/incentives/keeper/group_test.go:144]:
```suggestion
// As a result, we create this shared test case to avoid code duplication but
```

**Review by czarcas7ic — APPROVED** [review]
Again, really good job on these tests, quite clear despite all the overlapping logic

File: x/incentives/keeper/group.go [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+// - given pool IDs slice is empty or has 1 pool only
+// - fails to initialize gauge information for every pool ID
+// - fails to send coins from owner to the incentives module for the Group's Gauge
+// - fails to charge group creation fee
+// - fails to set the Group's Gauge to state
+func (k Keeper) CreateGroup(ctx sdk.Context, coins sdk.Coins, numEpochPaidOver uint64, owner sdk.AccAddress, poolIDs []uint64) (uint64, error) {
+	newGroup, err := k.createGroup(ctx, coins, numEpochPaidOver, owner, poolIDs)
+	if err != nil {
+		return 0, err
+	}
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply x/incentives/keeper/group.go:35]: Comment updated

**Pull request merged**